blog.qmd

```diff
@@ -0,0 +1,228 @@
+                  field_experience = c("fieldwork", "field experience", "field-experience")))
+```
+
+Armed with our dictionary, we're ready to search for the terms using the kwic() (key-word in context) function. This function takes in a corpus and a dictionary as inputs, along with a window parameter specifying the number of tokens before and after a keyword that we want to see for context.
+
+```{r}
+keywords <- kwic(corpus, dict, window = 10) %>% as_tibble()
+```
+
+```{r}
+#| echo: false
+keywords %>% 
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+Finally, we can join the results from the dictionary-based keyword in-context search back to the course level data and perform some wrangling to analyze the prevalence of work-based learning opportunities in Florida's community colleges.
+
+```{r}
+courses_with_keywords <-
+  keywords %>% 
+  left_join(courses, by = c("docname" = "doc_id")) %>%
+  mutate(sentence = glue("{pre} {keyword} {post}") |> str_trim() |> str_to_sentence()) %>% 
+  separate(docname, into = c("school", "course"), sep = "-", extra = 'merge'  ) %>% 
+  mutate(discipline = str_remove(discipline, ".* - ")) %>% 
+  select(school, discipline, course, statewide_course, degree_type, course_credits,  sentence, pattern) |>
+  arrange(school, pattern, discipline) %>% 
+  mutate(pattern = str_to_title(pattern) |> str_replace_all('_', '-'))
+```
+
+```{r}
+#| echo: false
+courses_with_keywords %>% 
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+```{r}
+freq_table <-
+  courses_with_keywords %>%
+  group_by(school, pattern) %>%
+  count() %>%
+  group_by(pattern) %>%
+  bind_rows(summarise(.,
+                      across(where(is.numeric), sum),
+                      across(where(is.character), ~ "total"))) %>% 
+  ungroup() %>%
+  pivot_wider(names_from = pattern, values_from = n) %>%
+  mutate(across(where(is.numeric), ~ replace_na(.x, 0))) %>% 
+  slice(n(), 1:n()-1)
+```
+
+```{r}
+#| echo: false
+freq_table %>%
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive()
+```
+
+## Conclusion
+
+In this post, we've demonstrated how to use quanteda to analyze course
+descriptions and identify work-based learning opportunities in community
+colleges. While our analysis focused on Florida, the same methods could be
+applied to other states or regions.
+
+Through this analysis, we've gained valuable insights into the prevalence of
+work-based learning in Florida's community colleges. We hope that our work can
+serve as a foundation for further research and policy discussions on this
+important topic.
+
+The power of quanteda lies not just in its ability to handle large text data, but also in its flexibility. It allows researchers to tailor their analysis to their specific needs, whether that's identifying key terms, comparing text documents, or exploring text patterns.
+
```

Comment:
I think this paragraph could be swapped with the previous one. I also think some mention of the comparative advantages of quanteda could be elaborated upon at the top of section 2 - why did we choose this instead of other packages? If it's the flexibility (which seems like a good answer!) I'd emphasize either up there or down here that this was an *exploratory* analysis so that flexibility was especially important